src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -143,4 +171,35 @@ public AsyncJsonTokenParser(
     return Futures.transformAsync(futureVerifiers, checkNullFunction, executor);
   }
 
+  /**
+   * Converts exceptions (when applicable) to {@link InvalidJsonTokenException}
+   * for better exception handling in the asynchronous parser.
+   *
+   * @return a failed {@link ListenableFuture}
+   */
+  private ListenableFuture getFailedFuture(Exception e) {
+    Throwable cause = e.getCause();
+    if (cause != null && InvalidJsonTokenException.class.isInstance(cause)) {
```

Comment:
Use instanceof rather than Class#isInstance

Reply:
Also `null instanceof` is always false so you can omit the first null check.

Reply:
Changed.